pkg/entities/message_repost_history.go

```diff
@@ -16,11 +17,23 @@ func (e *Entity) IsRepostableMessage(req request.CreateMessageRepostHistRequest)
 	return false
 }
 
-func (e *Entity) CreateRepostMessageHist(req request.CreateMessageRepostHistRequest, repostChannelID string) error {
-	return e.repo.MessageRepostHistory.CreateIfNotExist(model.MessageRepostHistory{
-		GuildID:         req.GuildID,
-		OriginMessageID: req.MessageID,
-		OriginChannelID: req.ChannelID,
-		RepostChannelID: repostChannelID,
-	})
+func (e *Entity) CreateRepostMessageHist(req request.CreateMessageRepostHistRequest, repostChannelID string) (*model.MessageRepostHistory, error) {
+	repostMsg, err := e.repo.MessageRepostHistory.GetByMessageID(req.GuildID, req.MessageID)
+	if err != nil && err != gorm.ErrRecordNotFound {
+		e.log.Error(err, "[entities.CreateRepostMessageHist] - failed to get repost message")
+		return nil, err
+	}
+	if err == gorm.ErrRecordNotFound {
+		err = e.repo.MessageRepostHistory.CreateIfNotExist(model.MessageRepostHistory{
+			GuildID:         req.GuildID,
+			OriginMessageID: req.MessageID,
+			OriginChannelID: req.ChannelID,
+			RepostChannelID: repostChannelID,
+		})
+		if err != nil {
+			return nil, err
+		}
```

Comment:
if `err == nil` here, we continue to jump to line 38 which will definitely return `nil, nil`.
(because when previous `err == gorm.ErrRecordNotFound` so repostMsg = nil)

suggest to update `CreateIfNotExist` to receive pointer as param
=> then we can have the model after creation and return here